src/library/Uno.Material/Styles/Controls/v2/TextBox.xaml

```diff
@@ -398,6 +399,13 @@
 										<Setter Target="Root.Background" Value="{ThemeResource FilledTextBoxBackgroundDisabled}" />
 										<Setter Target="DeleteButton.Foreground" Value="{ThemeResource FilledTextBoxDeleteButtonForegroundDisabled}" />
 										<Setter Target="IconPresenter.Foreground" Value="{ThemeResource TextBoxLeadingIconForegroundDisabled}" />
+
+										<!-- Header and Placeholder animations -->
+										<Setter Target="HeaderElement_CompositeTransform.TranslateY" Value="{Binding RelativeSource={RelativeSource TemplatedParent}, Converter={StaticResource HeaderCompositeTransformTranslateY}, TargetNullValue=0, FallbackValue=0}" />
+										<Setter Target="ContentElement_CompositeTransform.TranslateY" Value="{Binding RelativeSource={RelativeSource TemplatedParent}, Converter={StaticResource PlaceholderContentCompositeTransformTranslateY}, TargetNullValue=0, FallbackValue=0}" />
+										<Setter Target="PlaceholderElement_CompositeTransform.TranslateY" Value="{Binding RelativeSource={RelativeSource TemplatedParent}, Converter={StaticResource PlaceholderContentCompositeTransformTranslateY}, TargetNullValue=0, FallbackValue=0}" />
+										<Setter Target="HeaderElement_CompositeTransform.ScaleX" Value="{Binding RelativeSource={RelativeSource TemplatedParent}, Converter={StaticResource HeaderCompositeTransformScales}, TargetNullValue=1, FallbackValue=1}" />
+										<Setter Target="HeaderElement_CompositeTransform.ScaleY" Value="{Binding RelativeSource={RelativeSource TemplatedParent}, Converter={StaticResource HeaderCompositeTransformScales}, TargetNullValue=1, FallbackValue=1}" />
```

Comment:
Are these setters always being set to the same thing for each visual state? Could we maybe just set these once directly on the Transforms in the template? And if there are slight differences for specific visual states we could then add a Setter for the given VisualState?

cc: @Xiaoy312 thoughts?

Reply:
I've implemented your suggestion locally. But to make it work I needed to bring back the `VisualStateGroup x:Name="HeaderStates"`, which contains an `Empty` visual state and a `NotEmpty` visual state, to place the animations there, otherwise after animating the values would go back to the "normal" state and the header wouldn't keep up.

Also I needed to add those 5 `Setters` to the `Empty` visual state to have it working when a TextBox is created with `Text` for example, otherwise when removing the `Text` the header would keep up.

I'm wondering if this would still work if a TextBox is modified after it's creation, for example, an event that would add a `Header` or `PlaceholderText` to the TextBox, would it be dynamic to re-adjust those values since we wouldn't be using the visual states anymore?

@kazo0 @Xiaoy312

Reply:
## With code from master branch:
![txtb test master](https://github.com/user-attachments/assets/539cb925-3088-40b9-a489-542f9bfd2f55)


## With my local changes (mentioned on the previous comment):
![txtb test local](https://github.com/user-attachments/assets/9645279e-9c5f-4338-bc6b-682ace1ec4e1)

Reply:
We are keeping `Setter`s in all the visual states needed and we added `HeaderStates` to manage changes made via code.